version: '3.8'

services:
  db:
    image: postgres
    environment:
      POSTGRES_DB: "HexOcean"
      PGUSER: "postgres"
      POSTGRES_PASSWORD: "niema"
      POSTGRES_PORT: 5432
    volumes:
      - postgres-data:/var/lib/postgresql/data/
    healthcheck:
      test: ["CMD", "pg_isready", "-h", "localhost"]
      timeout: 10s
      retries: 5
    networks:
      - app-network
  web:
    build: .
    command: python manage.py runserver 0.0.0.0:8000
    volumes:
      - .:/code
    ports:
      - "8000:8000"
    depends_on:
      db:
        condition: service_healthy
    networks:
      - app-network

volumes:
  postgres-data:

networks:
  app-network:
    driver: bridge
